enhancements/cvo/object-removal-manifest-annotation.md

```diff
@@ -0,0 +1,187 @@
+```
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cluster-autoscaler-operator
+  namespace: openshift-machine-api
+  annotations:
+    release.openshift.io/delete: "true"
+...
+```
+Additional manifest properties such as `spec` may be set if convenient (e.g. because you are looking to make a minimal change vs. a previous version of the manifest), but those properties have no affect on manifests with the delete annotation.
+
+#### Story 2
+
+In release 4.5 two jobs, openshift-service-catalog-controller-manager-remover and openshift-service-catalog-apiserver-remover, were created to remove the Service Catalog.
+Now, for release 4.6, these jobs and all their supporting cluster objects must also be removed.
+This User Story shows how Service Catalog removal would have been executed had this enhancement been in place.
+The Service Catalog is composed of two components, the cluster-svcat-apiserver-operator and the cluster-svcat-controller-manager-operator.
+Each of these components use manifests for creation/update of the component's required resources: namespace, roles, operator deployment, etc.
+For example, the cluster-svcat-apiserver-operator has the following associated manifests:
+* 0000_50_cluster-svcat-apiserver-operator_00_namespace.yaml
+* 0000_50_cluster-svcat-apiserver-operator_02_config.crd.yaml
+* 0000_50_cluster-svcat-apiserver-operator_03_config.cr.yaml
+* 0000_50_cluster-svcat-apiserver-operator_03_configmap.yaml
+* 0000_50_cluster-svcat-apiserver-operator_03_version-configmap.yaml
+* 0000_50_cluster-svcat-apiserver-operator_04_roles.yaml
+* 0000_50_cluster-svcat-apiserver-operator_05_serviceaccount.yaml
+* 0000_50_cluster-svcat-apiserver-operator_06_service.yaml
+* 0000_50_cluster-svcat-apiserver-operator_07_deployment.yaml
+* 0000_50_cluster-svcat-apiserver-operator_08_cluster-operator.yaml
+
+Assuming all the above created resources are `namespace-scoped` and in the cluster-svcat-apiserver-operator namespace openshift-service-catalog-apiserver-operator, all could be deleted by simply deleting the namespace openshift-service-catalog-apiserver-operator.
+And, given this enhancement, the namespace delete would be done by adding the delete annotation to the manifest 0000_50_cluster-svcat-apiserver-operator_00_namespace.yaml.
+
+If any `cluster-scoped` resources do exist or resources in another namespace exist, each would have to be deleted explicitly using the same method of adding the delete annotation to the relevant manifest.
+If multiple delete manifests are required it is up to the developer to name the manifests such that deletions occur in the correct order.
+
+If resources external to kubernetes must be removed the developer must provide the means to do so. This could be in the form of a job or through modification of an operator to do the removals during it's finalization.
```

Comment:
Need to explain the process of modifying an existing operator to do this type of removal in more detail. We have to first update the operator in order to "get" this new delete logic and then do the deletes. I suppose this can be accomplished by proper manifest ordering.